app/frontend/wallet/crypto-provider-factory.js

```diff
@@ -0,0 +1,52 @@
+const Cardano = require('./cardano-wallet')
+const {ADALITE_CONFIG} = require('../config')
+const {LEDGER, TREZOR, MNEMONIC} = require('./constants')
+const derivationSchemes = require('./derivation-schemes')
+
+const CryptoProviderFactory = (() => {
+
+  const UnknownCryptoProviderError = (message) => {
+    const error = new Error(message)
+    error.name = 'UnknownCryptoProviderError'
+    return error
+  }
+
+  const getWallet = async (cryptoProvider, secret) => {
+    switch (cryptoProvider) {
```

Comment:
Note, we usually try to avoid switch. One idea might be

const configFactory = {
  [TREZOR]: () => {....},
  [LEDGER]: ...
}
const config = configFactory[provider]()
return await Cardano.CardanoWallet(config)
```
Looking at this, the whole function can be renamed to `walletConfigFactory` and would return only config, caller would do Cardano.CardanoWallet ...

Reply:
With this approach, we lose ability to send secret parameter, when the crypto provider is mnemonic, don't we?

Reply:
not really, functions have access to the outer scope in js